zipline/pipeline/loaders/events.py

```diff
@@ -36,196 +42,128 @@
 
 class EventsLoader(PipelineLoader):
     """
-    Abstract loader.
+    Base class for PipelineLoaders that supports loading the next and previous
+    value of an event field.
 
-    Does not currently support adjustments to the dates of known events.
+    Does not currently support adjustments.
 
     Parameters
     ----------
+    events : pd.DataFrame
+        A DatetimeIndexed DataFrame representing events (e.g. share buybacks or
+        earnings announcements) associated with particular companies.
+
+        ``events`` must contain at least three columns::
+            sid : int64
+                The asset id associated with each event.
+
+            event_date : datetime64[ns]
+                The date on which the event occurred.
+
+            timestamp : datetime64[ns]
+                The date on which we learned about the event. Must be greater
+                than or equal to the frame index value.
```

Comment:
If we require timestamp as a separate column, what is the DatetimeIndex for, and why does timestamp need to be greater than or equal to it?